src/scripts/main.js

```diff
@@ -1,3 +1,178 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+function formatSalary(cellValue) {
+  return parseFloat(cellValue.replace(/[$,]/g, ''));
+}
+
+function sortRows(index, isAscending, rowsArr) {
+  return rowsArr.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim();
+    const cellB = rowB.cells[index].textContent.trim();
+
+    if (index === 4) {
+      const salaryA = formatSalary(cellA);
+      const salaryB = formatSalary(cellB);
+
+      return isAscending ? salaryA - salaryB : salaryB - salaryA;
+    } else {
+      return isAscending
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    }
+  });
+}
+
+function updateTable(sortedRows) {
```

Comment:
The function `updateTable` is directly manipulating the DOM by setting `innerHTML` to an empty string and then appending child nodes. This could lead to performance issues. Consider creating a document fragment, appending the rows to it, and then appending the fragment to the table body.